### PR TITLE
[TASK] draw different colors on map and forcegraph for clients

### PR DESCRIPTION
**Pull request by genofire**
(just for discussion and performance analyse)
## Colors
wifi5: violet
other: orange
everything else (hopply wifi24): pink
## Map:
screenshot from 2017-10-26 18-19-09
## Forcegraph:
screenshot from 2017-10-26 18-19-24

### CODE REVIEW

**Comment by xf-** [comment]
Current local patch. Canvas color can't be trusted. Opacity changes (looks like a float).

```color.diff
5946b10b0ca25ce65fd91fafb367ee08d2495bd7
 lib/map.js             | 2 +-
 lib/map/clientlayer.js | 2 +-
 lib/utils/helper.js    | 8 +++++---
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/lib/map.js b/lib/map.js
index 58880ed..a44b12d 100644
--- a/lib/map.js
+++ b/lib/map.js
@@ -144,7 +144,7 @@ define(['map/clientlayer', 'map/labellayer', 'map/button', 'leaflet'],
         if (highlight !== undefined) {
           if (highlight.type === 'node' && nodeDict[highlight.o.node_id]) {
             m = nodeDict[highlight.o.node_id];
-            m.setStyle({ color: 'orange', weight: 10, fillOpacity: 1, opacity: 0.5, className: 'stroke-first' });
+            m.setStyle({ color: '#ad2358', weight: 8, fillOpacity: 1, opacity: 0.4, className: 'stroke-first' });
           } else if (highlight.type === 'link' && linkDict[highlight.o.id]) {
             m = linkDict[highlight.o.id];
             m.setStyle({ weight: 4, opacity: 1, dashArray: '5, 10' });
diff --git a/lib/map/clientlayer.js b/lib/map/clientlayer.js
index 4ae3333..9bc5281 100644
--- a/lib/map/clientlayer.js
+++ b/lib/map/clientlayer.js
@@ -45,7 +45,7 @@ define(['leaflet', 'rbush', 'helper'],
           return tile;
         }
 
-        var startDistance = 12;
+        var startDistance = 10;
 
         nodes.forEach(function (d) {
           var p = map.project([d.node.location.latitude, d.node.location.longitude]);
diff --git a/lib/utils/helper.js b/lib/utils/helper.js
index b3f47c2..6efd42f 100644
--- a/lib/utils/helper.js
+++ b/lib/utils/helper.js
@@ -160,6 +160,7 @@ define({
 
     var radius = 3;
     var a = 1.2;
+    var mode = 0;
 
     for (var orbit = 0, i = 0; i < node.clients; orbit++) {
       var distance = startDistance + orbit * 2 * radius * a;
@@ -167,12 +168,13 @@ define({
       var delta = node.clients - i;
 
       for (var j = 0; j < Math.min(delta, n); i++, j++) {
-        if (i >= (node.clients - node.clients_wifi5 - node.clients_other) && ctx.fillStyle !== 'rgba(147, 0, 220, 0.7)') {
+        if (mode === 0 && i >= node.clients_wifi24) {
+          mode = 2;
           ctx.fill();
           ctx.beginPath();
           ctx.fillStyle = 'rgba(147, 0, 220, 0.7)';
-        }
-        if (i >= (node.clients - node.clients_other) && ctx.fillStyle !== 'rgba(220, 128, 0, 0.7)') {
+        } else if (mode !== 1 && i >= (node.clients_wifi24 + node.clients_wifi5)) {
+          mode = 1;
           ctx.fill();
           ctx.beginPath();
           ctx.fillStyle = 'rgba(220, 128, 0, 0.7)';
```

**Comment by xf-** [comment]
```colors.diff
diff --git a/lib/about.js b/lib/about.js
index b23e640..ee36991 100644
--- a/lib/about.js
+++ b/lib/about.js
@@ -5,6 +5,18 @@ define(function () {
     this.render = function render(d) {
       d.innerHTML = _.t('sidebar.aboutInfo') +
 
+        '<p class="legend">' +
+        '<span class="legend-new"><span class="symbol"></span> ' + _.t('sidebar.nodeNew') + '</span>' +
+        '<span class="legend-online"><span class="symbol"></span> ' + _.t('sidebar.nodeOnline') + '</span>' +
+        '<span class="legend-offline"><span class="symbol"></span> ' + _.t('sidebar.nodeOffline') + '</span>' +
+        '</p>' +
+
+        '<p class="legend">' +
+        '<span class="legend-24ghz"><span class="symbol"></span> 2.4 Ghz ' + _.t('node.clients') + '</span>' +
+        '<span class="legend-5ghz"><span class="symbol"></span> 5 Ghz ' + _.t('node.clients') + '</span>' +
+        '<span class="legend-others"><span class="symbol"></span> ' + _.t('others') + ' ' + _.t('node.clients') + '</span>' +
+        '</p>' +
+
         '<h3>AGPL 3</h3>' +
 
         '<p>Copyright (C) Milan Pässler</p>' +
diff --git a/lib/legend.js b/lib/legend.js
index 530accf..5756dd7 100644
--- a/lib/legend.js
+++ b/lib/legend.js
@@ -32,15 +32,10 @@ define(['helper'], function (helper) {
 
       var p = document.createElement('p');
       p.classList.add('legend');
-      p.innerHTML = '<span class="legend-new"><span class="symbol"></span> ' + _.t('sidebar.nodeNew') + '</span>' +
-        '<span class="legend-online"><span class="symbol"></span> ' + _.t('sidebar.nodeOnline') + '</span>' +
-        '<span class="legend-offline"><span class="symbol"></span> ' + _.t('sidebar.nodeOffline') + '</span>';
-      el.appendChild(p);
-
-      p.appendChild(document.createElement('br'));
       p.appendChild(stats);
       p.appendChild(document.createElement('br'));
       p.appendChild(timestamp);
+      el.appendChild(p);
     };
 
     return self;
diff --git a/lib/utils/helper.js b/lib/utils/helper.js
index 6efd42f..e1b9fdf 100644
--- a/lib/utils/helper.js
+++ b/lib/utils/helper.js
@@ -172,12 +172,12 @@ define({
           mode = 2;
           ctx.fill();
           ctx.beginPath();
-          ctx.fillStyle = 'rgba(147, 0, 220, 0.7)';
+          ctx.fillStyle = 'rgba(227, 166, 25, 0.7)';
         } else if (mode !== 1 && i >= (node.clients_wifi24 + node.clients_wifi5)) {
           mode = 1;
           ctx.fill();
           ctx.beginPath();
-          ctx.fillStyle = 'rgba(220, 128, 0, 0.7)';
+          ctx.fillStyle = 'rgba(10, 156, 146, 0.7)';
         }
         var angle = 2 * Math.PI / n * j;
         var x = p.x + distance * Math.cos(angle + startAngle);
diff --git a/locale/en.json b/locale/en.json
index 02444e6..fb6f6a4 100644
--- a/locale/en.json
+++ b/locale/en.json
@@ -86,6 +86,6 @@
   "yes": "yes",
   "no": "no",
   "unknown": "unknown",
-  "others": "others",
+  "others": "other",
   "none": "none"
 }
diff --git a/scss/modules/_legend.scss b/scss/modules/_legend.scss
index 50e96c1..b0fa223 100644
--- a/scss/modules/_legend.scss
+++ b/scss/modules/_legend.scss
@@ -17,6 +17,12 @@ header {
 }
 
 .legend {
+  span {
+    &:not(:first-child) {
+      margin-left: 1em;
+    }
+  }
+
   .symbol {
     border-radius: 50%;
     display: inline-block;
@@ -24,6 +30,7 @@ header {
     vertical-align: -5%;
     width: 1em;
   }
+
 }
 
 // Dot looks compared to thin font a bit darker - lighten it 10%
@@ -45,7 +52,20 @@ header {
   }
 }
 
-.legend-online,
-.legend-offline {
-  margin-left: 1em;
+.legend-24ghz {
+  .symbol {
+    background-color: $color-24ghz;
+  }
+}
+
+.legend-5ghz {
+  .symbol {
+    background-color: $color-5ghz;
+  }
+}
+
+.legend-others {
+  .symbol {
+    background-color: $color-others;
+  }
 }
diff --git a/scss/modules/_variables.scss b/scss/modules/_variables.scss
index 6e405eb..36f11ad 100644
--- a/scss/modules/_variables.scss
+++ b/scss/modules/_variables.scss
@@ -11,6 +11,10 @@ $color-online: #1566a9 !default;
 $color-offline: #cf3e2a !default;
 $color-unseen: #d89100 !default;
 
+$color-24ghz: $color-primary !default;
+$color-5ghz: #e3a619 !default;
+$color-others: #0a9c92 !default;
+
 $color-map-background: #f8f4f0 !default;
 
 $font-family: Assistant, sans-serif !default;
```

**Comment by genofire** [comment]
screenshot from 2017-10-28 02-30-34
screenshot from 2017-10-28 02-30-46

File: locale/fr.json [hunk 1]
```diff
@@ -44,9 +44,9 @@
     "clients": "avec %{smart_count} client |||| avec %{smart_count} clients",
     "gateway": "sur %{smart_count} passerelle |||| sur %{smart_count} passerelles",
     "lastUpdate": "Dernière actualisation",
-    "nodeNew": "Nœud est nouveau",
-    "nodeOnline": "Nœud est en ligne",
-    "nodeOffline": "Nœud hors ligne",
+    "nodeNew": "nouveau",
```
Comment (xf-) [inline locale/fr.json]: done via crowdin

File: scss/modules/_infobox.scss [hunk 2]
```diff
@@ -7,6 +7,14 @@
       text-align: center;
     }
 
+    .symbol {
```
Comment (xf-) [inline scss/modules/_infobox.scss]: nogo - copy css block - we should move the current one (remove nesting) `.symbol` is now global anyway

File: lib/about.js [hunk 3]
```diff
@@ -4,7 +4,17 @@ define(function () {
   return function () {
     this.render = function render(d) {
       d.innerHTML = _.t('sidebar.aboutInfo') +
+        '<p class="legend">' + _.t('node.nodes') + ': ' +
```
Comment (xf-) [inline lib/about.js]: good idea - i would use a h4
Indents are from js and not html

**Review by xf- — CHANGES_REQUESTED** [review]

**Comment by xf-** [comment]
```foo.patch
diff --git a/lib/about.js b/lib/about.js
index 97cec77..8d48ca6 100644
--- a/lib/about.js
+++ b/lib/about.js
@@ -4,16 +4,17 @@ define(function () {
   return function () {
     this.render = function render(d) {
       d.innerHTML = _.t('sidebar.aboutInfo') +
-        '<p class="legend">' + _.t('node.nodes') + ': ' +
-          '<span class="legend-new"><span class="symbol"></span> ' + _.t('sidebar.nodeNew') + '</span>' +
-          '<span class="legend-online"><span class="symbol"></span> ' + _.t('sidebar.nodeOnline') + '</span>' +
-          '<span class="legend-offline"><span class="symbol"></span> ' + _.t('sidebar.nodeOffline') + '</span>' +
+        '<h4>' + _.t('node.nodes') + '</h4>' +
+        '<p class="legend">' +
+        '<span class="legend-new"><span class="symbol"></span> ' + _.t('sidebar.nodeNew') + '</span>' +
+        '<span class="legend-online"><span class="symbol"></span> ' + _.t('sidebar.nodeOnline') + '</span>' +
+        '<span class="legend-offline"><span class="symbol"></span> ' + _.t('sidebar.nodeOffline') + '</span>' +
         '</p>' +
-
-        '<p class="legend">' + _.t('node.clients') + ': ' +
-          '<span class="legend-24ghz"><span class="symbol"></span> 2.4 Ghz</span>' +
-          '<span class="legend-5ghz"><span class="symbol"></span> 5 Ghz</span>' +
-          '<span class="legend-others"><span class="symbol"></span> ' + _.t('others') + '</span>' +
+        '<h4>' + _.t('node.clients') + '</h4>' +
+        '<p class="legend">' +
+        '<span class="legend-24ghz"><span class="symbol"></span> 2.4 Ghz</span>' +
+        '<span class="legend-5ghz"><span class="symbol"></span> 5 Ghz</span>' +
+        '<span class="legend-others"><span class="symbol"></span> ' + _.t('others') + '</span>' +
         '</p>' +
         '<h3>AGPL 3</h3>' +
 
diff --git a/lib/infobox/node.js b/lib/infobox/node.js
index c8b3ff5..ab7c7c6 100644
--- a/lib/infobox/node.js
+++ b/lib/infobox/node.js
@@ -79,24 +79,21 @@ define(['sorttable', 'snabbdom', 'd3-interpolate', 'moment', 'helper'],
           [
             d.clients_wifi24,
             V.h('br'),
-            V.h('span', { props: { className: 'symbol'}}),
-            '2,4 Ghz'
+            V.h('span', { props: { className: 'symbol', title: '2,4 Ghz'}})
           ]),
         V.h('span',
           { props: { className: 'legend-5ghz'}},
           [
             d.clients_wifi5,
             V.h('br'),
-            V.h('span', { props: { className: 'symbol'}}),
-            '5 Ghz'
+            V.h('span', { props: { className: 'symbol', title: '5 Ghz'}})
           ]),
         V.h('span',
           { props: { className: 'legend-others'}},
           [
             d.clients_other,
             V.h('br'),
-            V.h('span', { props: { className: 'symbol'}}),
-            _.t('others')
+            V.h('span', { props: { className: 'symbol', title: _.t('others')}})
           ])
       ];
 
diff --git a/scss/modules/_infobox.scss b/scss/modules/_infobox.scss
index 0385738..5bf25e3 100644
--- a/scss/modules/_infobox.scss
+++ b/scss/modules/_infobox.scss
@@ -7,14 +7,6 @@
       text-align: center;
     }
 
-    .symbol {
-      border-radius: 50%;
-      display: inline-block;
-      height: .6em;
-      margin-right: 2px;
-      width: .6em;
-    }
-
     .ion-people {
       font-size: 1.5em;
     }
diff --git a/scss/modules/_legend.scss b/scss/modules/_legend.scss
index 41c2fe1..5a1b650 100644
--- a/scss/modules/_legend.scss
+++ b/scss/modules/_legend.scss
@@ -23,13 +23,14 @@ header {
     }
   }
 
-  .symbol {
-    border-radius: 50%;
-    display: inline-block;
-    height: 1em;
-    vertical-align: -5%;
-    width: 1em;
-  }
+}
+
+.symbol {
+  border-radius: 50%;
+  display: inline-block;
+  height: 1em;
+  vertical-align: -5%;
+  width: 1em;
 }
 
 // Dot looks compared to thin font a bit darker - lighten it 10%
```

**Comment by genofire** [comment]
screenshot from 2017-10-29 01-11-55

File: locale/de.json [hunk 4]
```diff
@@ -44,9 +44,9 @@
     "clients": "mit %{smart_count} Nutzer |||| mit %{smart_count} Nutzern",
     "gateway": "auf %{smart_count} Gateway |||| auf %{smart_count} Gateways",
     "lastUpdate": "Letzte Aktualisierung",
-    "nodeNew": "Knoten ist neu",
-    "nodeOnline": "Knoten ist online",
-    "nodeOffline": "Knoten ist offline",
+    "nodeNew": "neu",
```
Comment (xf-) [inline locale/de.json]: crowdin - won't upload targets

**Review by xf- — CHANGES_REQUESTED** [review]

**Review by xf- — APPROVED** [review]

**Pull request merged**